doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,634 @@
+
+Feel free to skip it, unless you are familiar with the implementation of the
+algorithm, or you discover some counter-intuitive behavior while writing the
+examples for the function.
+
+This section follows the same format as the extended summary section.
+
+Section 7: Examples
+~~~~~~~~~~~~~~~~~~~
+
+This is one of the most important sections of a docstring, even if it is
+placed in the last position. As often, people understand concepts better
+with examples, than with accurate explanations.
+
+Examples in docstrings are also unit tests, and besides illustrating the
+usage of the function or method, they need to be valid Python code, that in a
+deterministic way returns the presented output.
+
+They are presented as a session in the Python terminal. `>>>` is used to
+present code. `...` is used for code continuing from the previous line.
+Output is presented immediately after the last line of code generating the
+output (no blank lines in between). Comments describing the examples can
+be added with blank lines before and after them.
+
+The way to present examples is as follows:
+
+1. Import required libraries
+
+2. Create the data required for the example
+
+3. Show a very basic example that gives an idea of the most common use case
+
+4. Add commented examples that illustrate how the parameters can be used for
+   extended functionality
+
+A simple example could be:
+
+.. code-block:: python
+
+    class Series:
+        def head(self, n=5):
+            """Return the first elements of the Series.
+
+            This function is mainly useful to preview the values of the
+            Series without displaying the whole of it.
+
+            Parameters
+            ----------
+            n : int
+                Number of values to return.
+
+            Return
+            ------
+            pandas.Series
+                Subset of the original series with the n first values.
+
+            See Also
+            --------
+            tail : Return the last n elements of the Series.
+
+            Examples
+            --------
+            >>> import pandas
+            >>> s = pandas.Series(['Ant', 'Bear', 'Cow', 'Dog', 'Falcon',
+            ...                    'Lion', 'Monkey', 'Rabbit', 'Zebra'])
+            >>> s.head()
+            0   Ant
+            1   Bear
+            2   Cow
+            3   Dog
+            4   Falcon
+            dtype: object
+
+            With the `n` parameter, we can change the number of returned rows:
+
+            >>> s.head(n=3)
+            0   Ant
+            1   Bear
+            2   Cow
+            dtype: object
+            """
+            return self.iloc[:n]
+
+Conventions for the examples
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+   numpydoc recommends avoiding "obvious" imports and importing them with
+   aliases, so for example `import numpy as np`. While this is now an standard
+   in the data ecosystem of Python, it doesn't seem a good practise, for the
+   next reasons:
```

Comment:
Fully acknowledging that I'm bike-shedding...but I really disagree with this.  Personally think aliasing is a great compromise between `import *` (like R, etc), while still having enough brevity for interactive workflows.  Also conflicts with majority of pandas/numpy code in the wild.  

I also agree with the numpydoc suggestion to avoid obvious imports - I suspect the first most common use of docstrings is inside a repl/notebook/etc - showing the imports adds noise in that context.

Reply:
About the aliasing (`import pandas as pd`, `import numpy as np`), I agree with @chris-b1. It's something new users will have to learn indeed, but it's something they will *need* to learn anyhow, as almost *any* code you will see online uses those aliases.

 > Also conflicts with majority of pandas/numpy code in the wild.

That's not my perception, but maybe I am a bit biased by my environment :)

About whether we want to show the imports or not, here I am more open to be convinced otherwise, although I also find having those two imports everywhere adds noise.